### PR TITLE
Make it easier to use ILMerge console app via NuGet and MSBuild 

### PR DESCRIPTION
**Pull request by natemcmaster**
Changes:
 - Upgraded two projects to VS 2017: ILMerge and ILMerge.Tests.
 - Added a nuspec so package re-generates on build.
 - Updated README with example of usage.
 - Added MSBuild properties file to NuGet package that sets the property ILMergeConsolePath to the location of ILMerge.exe.

This takes care of item 2 from https://github.com/Microsoft/ILMerge/issues/9#issuecomment-296254866

cc @mike-barnett

### CODE REVIEW

File: ILMerge/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,57 +0,0 @@
-using System.Reflection;
-using System.Runtime.CompilerServices;
-
-//
-// General Information about an assembly is controlled through the following 
-// set of attributes. Change these attribute values to modify the information
-// associated with an assembly.
-//
-[assembly: AssemblyTitle("ILMerge")]
-[assembly: AssemblyDescription("A tool for merging multiple .NET assemblies into a single assembly.")]
```
Comment (natemcmaster) [inline ILMerge/AssemblyInfo.cs:10]: FYI - the contents of this file are automatically generated on-build by VS 2017.

**Comment by natemcmaster** [comment]
By the way, I noticed that in 2.14, the package only includes ILMerge.exe. I assume this file has  System.Compiler.dll merged into it. I left them separate for now to keep build as simple as possible. Is this okay? Or should we run ILMerge to combine the files?

Current layout:
image

**Comment by mike-barnett** [comment]
Yes, I had always used ILMerge on itself to simplify deployment for people: they could just copy the one file to wherever they needed to use it. However, I never made it a post-build step to simplify debugging. I'm fine with leaving them as two files. What do you think?

**Comment by natemcmaster** [comment]
Yeah, that makes sense. I've added a post-build step to do this.

**Comment by natemcmaster** [comment]
Ping @mike-barnett can you take a look at this when you have a minute? It should be ready to merge.

**Comment by natemcmaster** [comment]
Thanks @mike-barnett !

**Comment by moh-hassan** [comment]
**@natemcmaster** 
Is there a wiki document to describe how to use ilmerge from msbuild.
Can you post how to use.

**Pull request merged**